obspy/taup/velocity_model.py

```diff
@@ -484,6 +485,125 @@ def readTVelFile(cls, filename):
                              cls.default_cmb, cls.default_iocb, 0,
                              maxRadius, True, layers)
 
+    @classmethod
+    def readNDFile(cls, filename):
+        """
+        This method reads in a velocity model from a "nd" ASCII text file, the
+        format used by Xgbm. The name of the model file for model "modelname"
+        should be "modelname.nd".
+        The format of the file is:
+        depth pVel sVel Density Qp Qs
+        depth pVel sVel Density Qp Qs
+        . . . with each major boundary separated with a line with "mantle",
+        "outer-core" or "inner-core". "moho", "cmb" and "icocb" are allowed
+        as synonyms respectively.
+
+        This feature makes phase interpretation much easier to
+        code. Also, as they are not needed for travel time calculations, the
+        density, Qp and Qs may be omitted.
+
+        The velocities are assumed to be linear between sample points. Because
+        this type of model file doesn't give complete information we make the
+        following assumptions:
+
+        modelname - from the filename, with ".nd" dropped, if present
+
+        radiusOfEarth - the largest depth in the model
+
+        Comments are allowed. # signifies that the rest of the
+        line is a comment.  If # is the first character in a line, the line is
+        ignored
+
+        :param filename: The name of the file to read.
+        :type filename: str
+
+        :raises ValueError: If model file is in error.
+        """
+        mohoDepth = cls.default_moho
+        cmbDepth = cls.default_cmb
+        iocbDepth = cls.default_iocb
+
+        # Read all lines from file to enable identifying top and bottom values
+        # for each layer and find named discontinuities if present        
+        with open(filename) as modfile:
+            lines = modfile.readlines()
+
+        # Loop through to fill data array and locate named discontinuities
+        ii = 0
+        for line in lines:
+            # Strip off anything after '#'
+            line = line.split('#')[0]
+            if len(line.split()) == 0:  # Skip empty or comment lines
+                continue
+            if ii == 0:
+                data = []
+                for item in line.split():
+                    data.append(float(item))
+                data = np.array(data)
+                ii = ii + 1
+            else:
+                if re.match('[a-zA-Z]', line.split()[0]):
```

Comment:
Yea my comments are really on point today ;-) Sorry about that...

After looking at it a bit more careful: the regex will only look at the first character so it is more or less equivalent with `line.split()[0][0].isalpha()`.

It might better to check the amount of components the line splits into - its only a named discontinuity if it splits in one item. Then it will also work for discontinuities that are named for example `440` which people might use.